### ISSUE TITLE
Add version of runWithWarningsSuppressed for synchronous functions

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5950]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5950|
|Type|enhancement|
|Created|2012-09-06 17:36:03Z|
|Branch|https://github.com/twisted/twisted/tree/sync-suppressed-warnings-5950|

`SynchronousTestCase` needs a version of `runWithWarningsSuppressed` that doesn't depend on Deferreds.

<details><summary>Searchable metadata</summary>

```
trac-id__5950 5950
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_sync_suppressed_warnings_5950 branches/sync-suppressed-warnings-5950
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1346952963000000 1346952963000000
changetime__1346957655000000 1346957655000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35538]) Branching to 'sync-suppressed-warnings-5950'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review.

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/sync-suppressed-warnings-5950 will run soon.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [35546]) Merge sync-suppressed-warnings-5950: Add synchronous runWithWarningSuppressed function.

Author: itamar
Review: exarkun
Fixes: [#5950](https://github.com/twisted/twisted/issues/5950)

Another step on the road to a non-reactor-requiring SynchronousTestCase - a version of runWithWarningSuppressed that is not aware of Defereeds. And has tests.
